gateway/dataimpl/server_v1/errorhandler.go

```diff
@@ -565,3 +565,9 @@ func (e ErrorHandler) NewGenericStatus(err error) *status.Status {
 
 	return e.NewUnknownStatus(err)
 }
+
+func (e ErrorHandler) NewInvalidArgumentStatus(baseErr error) *status.Status {
```

Comment:
If you look at the other cases of us using codes.InvalidArgument, we directly create the error as InvalidArgument doesn't really contain any contextual information except through it's string (such that every generated code has a unique string message associated with it).  Could you do the same thing with these as well?  This will make debugging why the invalid argument occurred a lot easier.

Reply:
Done I think